agent/pkg/chassis/registry/registry.go

```diff
@@ -175,7 +175,12 @@ func parseServiceURL(serviceURL string) (string, string, int, error) {
 func getPortAndProtocol(svc *v1.Service, svcPort int) (targetPort int, protocol string) {
 	for _, p := range svc.Spec.Ports {
 		if p.Protocol == "TCP" && int(p.Port) == svcPort {
-			protocol = strings.Split(p.Name, "-")[0]
+			pro := strings.Split(p.Name, "-")
+			if len(pro) >= 2 {
+				protocol = pro[0]
```

Comment:
I think we need to consider the validity of the value `pro`. EdgeMesh only support `http`, `tcp` now. If the value of `pro` is not valid, we can fetch the `p.Protocol` or just set the `tcp`

Reply:
yes, the value of `pro` needs to check. And the `websocket`、`grpc` protocol  have any plan to support it?

Reply:
Websocket is support by `http` protocol, [link](https://github.com/kubeedge/edgemesh/blob/main/examples/websocket.yaml). The grpc will support if it is necessary.